### ISSUE TITLE
Some arrays' contents deleted before reaching serialize

### ISSUE THREAD

**Issue by tannerlinsley** [js-data/js-data#74]
I'm `save`ing this object below (with items in every `group` field), and when it comes back, I found that every `groups` array but the first is empty when going out to the server like so:

```
{
    "id": 50,
    "name": "something",
    "integrations": {
        "rankings": {
            "keywords": [{
                "id": "6129495323884985535",
                "keyword": "Redrock",
                "engine": "google",
                "locale": "es_us",
                "geo": "",
                "device": "d",
                "groups": [
                    "Test"
                ]
            }, {
                "id": "10229707372952463379",
                "keyword": "Redrock",
                "engine": "google",
                "locale": "es_us",
                "geo": "Ogden",
                "device": "d",
                "groups": []
            }, {
                "id": "11044795523948746316",
                "keyword": "Redrock",
                "engine": "google",
                "locale": "es_us",
                "geo": "Kearns",
                "device": "d",
                "groups": []
            }, {
                "id": "9121884036407653290",
                "keyword": "Redrock",
                "engine": "google",
                "locale": "en_us",
                "geo": "",
                "device": "d",
                "groups": []
            }, {
                "id": "6149529768597086244",
                "keyword": "Redrock",
                "engine": "google",
                "locale": "en_us",
                "geo": "Moab",
                "device": "d",
                "groups": []
            }, {
                "id": "969714134800755271",
                "keyword": "Redrock",
                "engine": "google",
                "locale": "en_us",
                "geo": "Lehi",
                "device": "d",
                "groups": []
            }]
        }
    }
}


```

The furthest back I can trace it is to `serialize`, in which they have already been deleted.  So somewhere between my `save` and the `serialize` hook, they're getting affected...

**Comment by jmdobry** [comment]
Please add more information:
- good - Your versions of Angular, JSData, etc, relevant console logs/error, code examples that revealed the issue
- better - A [plnkr](http://plnkr.co/), [fiddle](http://jsfiddle.net/), or [bin](http://jsbin.com/?html,output) that demonstrates the issue
- best - A Pull Request that fixes the issue, including test coverage for the issue and the fix

**Comment by jmdobry** [comment]
Fixed in 1.8.0
